### PR TITLE
graph sdk: design doc

### PR DESCRIPTION
**Pull request by aramikm**
# Goal
The goal of this PR is provide a design doc for graph sdk implementation.

Closes issue #1146

### CODE REVIEW

File: designdocs/graph_sdk.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+| Name  | Description |
+| ------------- | ------------- |
+| **Initialise**  | Creates in memory graph structure for a desired MSA.  |
+| **Import**  | Import the blob from frequency into Graph SDK for desired MSA.  |
+| **Update**  | Changes the current in-memory graph structure with incoming updates.  |
+| **Get Graph**  | Exposes current state of in-memory graph to the consumer of the SDK.  |
+| **Has Updates**  | Determines if the applied changes created some updates in graph that needs to be persisted on Frequency.  |
+| **Calculate Updates**  | Applies the graph changes and generates optimized blobs to be applied to Frequency  |
+| **Persist**  | This should be called after successful update of Frequency to remove tracking of persisted updates.  |
```
Comment (saraswatpuneet) [inline designdocs/graph_sdk.md:36]: Did not really get the idea behind persistence, needs a bit rewording here

File: designdocs/graph_sdk.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+- Define interface of Graph SDK.
+- Define the main concepts and entities in Graph SDK.
+- Define the algorithms to optimize the output regarding Frequency.
+
+## Operations
+Following is a list of desired operations in this SDK:
+
+| Name  | Description |
+| ------------- | ------------- |
+| **Initialise**  | Creates in memory graph structure for a desired MSA.  |
```
Comment (enddynayn) [inline designdocs/graph_sdk.md:30]: Initialize

File: designdocs/graph_sdk.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+| **Import**  | Import the blob from frequency into Graph SDK for desired MSA.  |
```
Comment (enddynayn) [inline designdocs/graph_sdk.md:31]: maybe imports multiple blobs?

File: designdocs/graph_sdk.md (hunk 1 above)
Comment (JoeCap08055) [reply designdocs/graph_sdk.md:36]: I agree. Perhaps we need an additional paragraph before this to describe the SDK-in-memory <-> persistence-on-chain model.

Comment (aramikm) [reply designdocs/graph_sdk.md]: I'm removing the **persist** for now to simplify the flow. We might need to add it later based on how providers would prefer to interact with the sdk.

**Review by JoeCap08055 — APPROVED** [review]

**Comment by wesbiggs** [comment]
What do you think of using dsnp_user_id instead of msa_id? The former is the DSNP concept, the latter is a Frequency-specific one.

**Pull request merged**